testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -1813,6 +1844,372 @@ fn size_overflow_unconfirmed_microblocks_integration_test() {
     channel.stop_chains_coordinator();
 }
 
+// mine a stream of microblocks, and verify that the miner won't let us overflow the size
+#[test]
+#[ignore]
+fn size_overflow_unconfirmed_stream_microblocks_integration_test() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    let mut small_contract = "(define-public (f) (ok 1))".to_string();
+    for _i in 0..((1024 * 1024 + 500) / 3) {
+        small_contract.push_str(" ");
+    }
+
+    let spender_sks: Vec<_> = (0..25)
+        .into_iter()
+        .map(|_| StacksPrivateKey::new())
+        .collect();
+    let spender_addrs: Vec<PrincipalData> = spender_sks.iter().map(|x| to_addr(x).into()).collect();
+
+    let txs: Vec<Vec<_>> = spender_sks
+        .iter()
+        .map(|spender_sk| {
+            let mut ret = vec![];
+            for i in 0..1 {
+                let tx = make_contract_publish_microblock_only(
+                    spender_sk,
+                    i as u64,
+                    600000,
+                    &format!("small-{}", i),
+                    &small_contract,
+                );
+                ret.push(tx);
+            }
+            ret
+        })
+        .collect();
+
+    let flat_txs: Vec<_> = txs.iter().fold(vec![], |mut acc, a| {
```

Comment:
There's a [`flat_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flat_map) function that could be helpful here.